bohra/modules/ngmaster/main.nf

```diff
@@ -0,0 +1,42 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process NGMASTER {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:meta.id, publish_id:meta.id) }
+    
+    cpus options.args2// args2 needs to be cpus for shovill
+    cache 'lenient'
+    errorStrategy 'ignore'
+    
+    if ( params.enable_conda ) {
+        if (file("${params.conda_path}").exists()) {
+            conda "${params.conda_path}/bohra-ngmaster"
+        } else {
+            conda 'ngmaster csvtk'
+        }
+       
+    } else {
+        conda null
+    }
+    input:
+    tuple val(meta), path(contigs), val(species)
+
+    output:
+    tuple val(meta), path('typer.txt'), emit: typer
+    
+
+    script:
+    """
+    echo -e Isolate'\n'${meta.id} >> tmp.tab
+    ngmaster  $contigs  > ngmaster.tab
+    paste tmp.tab ngmaster.tab | csvtk -t cut -f -ID> typer.txt
```

Comment:
- [ ] Confirm `typer.txt` is created correctly from `ngmaster` output.

Reply:
@stroehleina  the conda installation has ID... maybe we/you could create a new conda release? The we can update..

Reply:
When I run (as in your bohra-install.sh):

`mamba create -y -n bohra-ngmaster ngmaster csvtk`
`mamba activate bohra-ngmaster`
`/home/astroehlein/.conda/envs/bohra-ngmaster/bin/ngmaster --test`

I get:

```
Running ngmaster.py on test example (NG-MAST 4186 / NG-STAR 231) ...
FILE    SCHEME  NG-MAST/NG-STAR porB_NG-MAST    tbpB    penA    mtrR    porB_NG-STAR    ponA    gyrA    parC    23S
/home/astroehlein/.conda/envs/bohra-ngmaster/lib/python3.9/site-packages/ngmaster/test/test.fa  ngmaSTar        4186/231        2569    241     23      42100      100     10      2       100
... Test successful.
```
`/home/astroehlein/.conda/envs/bohra-ngmaster/bin/ngmaster --version`
```
ngmaster 1.0.0
```

So for a fresh install (`bohra init`) you should get the latest (1.0.0) version which has the updated headers?

Reply:
See release [here](https://github.com/MDU-PHL/ngmaster/releases/tag/v1.0.0) and on [conda](https://anaconda.org/bioconda/ngmaster).

Reply:
@stroehleina the ngmaster install is a challenge and i don't want to tie to specific  version - just in case it is not expected so i have used `csvtk cut -f -2` which will remove whatever id col there is after adding in the is `Isolate` column